### DISCUSSION TITLE
How to get `&db_conn` in `main()`?

### DISCUSSION THREAD

**Discussion by Glitchy-Tozier** [rwf2/Rocket#2326] [Questions] [answered]
I'm trying to set up automatic running of migrations. ([See here](https://github.com/diesel-rs/diesel/issues/559))

Is there a way to get access to `&db_conn` inside `main()` (or `launch()`)? I think I'm looking for the same thing that is accessible in in routes such as this one:

```rs
/// Returns everything currently stored in the database.
#[get("/get_db")]
pub(crate) async fn get_db(conn: DbConn) -> DbData {
    conn.run(
        |c| {
            ...
            results
        },
    )
    .await
}

```

**Comment by Glitchy-Tozier** [comment, marked as answer]
I managed to obtain the DB-connection by using [this method](https://github.com/diesel-rs/diesel/blob/v1.4.4/examples/postgres/getting_started_step_1/src/lib.rs):

```rs
#[macro_use]
extern crate diesel;
extern crate dotenv;

pub mod models;
pub mod schema;

use diesel::prelude::*;
use dotenv::dotenv;
use std::env;

pub fn establish_connection() -> PgConnection {
    dotenv().ok();

    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
    PgConnection::establish(&database_url)
        .unwrap_or_else(|_| panic!("Error connecting to {}", database_url))
}
```

**Comment by SergioBenitez** [comment]
The appropriate approach is the one in the database example: https://github.com/SergioBenitez/Rocket/blob/master/examples/databases/src/sqlx.rs. That is, using a fairing.
